dynamics/wind_model.py

```diff
@@ -0,0 +1,201 @@
+
+    projection = (
+        (unit_wind_vector[0] * unit_car_vector[0] + unit_wind_vector[1] * unit_car_vector[1]) * unit_car_vector[0],
+        (unit_wind_vector[0] * unit_car_vector[0] + unit_wind_vector[1] * unit_car_vector[1]) * unit_car_vector[1]
+                )
+    unit_parallel_wind_component = projection
+
+    print("Unit Wind Vector:", unit_wind_vector)
+    print("Unit Car Vector:", unit_car_vector)
+    print("Unit Proj Vector:", unit_parallel_wind_component)
+    plt.quiver(x, y, unit_wind_vector[0], unit_wind_vector[1], color='b', units='xy', scale=1, label="Wind Direction")
+    plt.quiver(x, y, unit_car_vector[0], unit_car_vector[1], color='r', units='xy', scale=1, label="Car Direction")
+    plt.quiver(x, y, unit_parallel_wind_component[0], unit_parallel_wind_component[1], color='g', units='xy', scale=1,
+               label="Projected Wind")
+    plt.title('Vector Plot')
+
+    # set the x-limits and y-limits
+    plt.xlim(-1.5, 1.5)
+    plt.ylim(-1.5, 1.5)
+
+    # Show plot
+    plt.legend(loc="upper left")
+    plt.grid()
+    plt.show()
+
+
+def wind_model_main(coordinates_list):
+    """
+    Calculates the drag of the car for a series of two points, drag is calculated based on wind data at 2nd point
+    @param coordinates_list: list of length 2 containing tuples containing coordinates Ex: [(lat1, long1), (lat2, long2)]
+    @return: Drag force in N
+
+    """
+    # Generate vector modelling car's direction of travel
+    car_vector = gen_car_vector(coordinates_list)
+
+    # Get wind data from API for a given coordinate point
+    current_point = coordinates_list[1]
+    weather_api_result = []
+    weather_data = get_weather(str(current_point[0]), str(current_point[1]), weather_api_result)
+
+    # Parse down to relevant data
+    # For reference array structure is as follows: ['Latitude', 'Longitude',
+    # 'Temperature(C)', 'Wind Speed (m/s)',  'Wind Direction', 'Weather', 'Weather Description',
+    #  Pressure (hPa), 'Precipitation (mm)']
+    wind_speed = weather_data[0][3]
+    wind_direction = weather_data[0][4]
+    pressure = weather_data[0][7] * 100  # Convert hPa to Pa
+    temperature = weather_data[0][2] + 273  # Convert to absolute temperature
+
+    # Generate a wind vector and solve for component of wind parallel to car's direction of travel
+    wind_vector = gen_wind_vector(wind_speed, wind_direction)
+    parallel_wind_component = get_wind_component(wind_vector, car_vector)
+    parallel_wind_component_magnitude = math.sqrt(parallel_wind_component[0] * parallel_wind_component[0] +
+                                                  parallel_wind_component[1] * parallel_wind_component[1])
+
+    # If wind component is in same direction of car, must subtract wind component from car speed
+    if ((parallel_wind_component[0] * car_vector[0]) >= 0) or ((parallel_wind_component[1] * car_vector[0]) >= 1):
+        parallel_wind_component_magnitude = parallel_wind_component_magnitude * -1
```

Comment:
this should work, right?
```suggestion
        parallel_wind_component_magnitude *= -1
```

Reply:
Yep, will change this!